synapseclient/client.py

```diff
@@ -3075,6 +3075,65 @@ def getColumns(self, x, limit=100, offset=0):
         else:
             ValueError("Can't get columns for a %s" % type(x))
 
+    def create_snapshot(self, table, comment=None, label=None, activity=None):
+        """Creates Table or EntityView snapshots
+
+        :param table:  The schema of the Table
+        :param comment:  Optional snapshot comment.
+        :param label:  Optional snapshot label.
+        :param activity:  Optional activity ID applied to snapshot version.
+        """
+        ent = self.get(id_of(table), downloadFile=False)
+        if isinstance(ent, EntityViewSchema):
+            self._async_table_update(table, create_snapshot=True,
+                                     comment=comment, label=label,
+                                     activity=activity)
+        elif isinstance(ent, Schema):
+            self._create_table_snapshot(table, comment=comment, label=label,
+                                        activity=activity)
+        else:
+            raise ValueError("This function only accepts Synapse ids of Tables or EntityViews")
+
+    def _create_table_snapshot(self, table, comment=None, label=None, activity=None):
+        """Creates Table snapshot
+
+        :param table:  The schema of the Table
+        :param comment:  Optional snapshot comment.
+        :param label:  Optional snapshot label.
+        :param activity:  Optional activity ID applied to snapshot version.
+
+        :return:  Snapshot Response
+        """
+        snapshot_body = {"snapshotComment": comment,
```

Comment:
Could we only include the dictionary keys if the value is not default/None. We aren't consistent about that in the client now, but I'm trying to move towards absence of a key for a default/null value since I think there is less ambiguity about what that means to the backend.

Reply:
Thanks @jkiang13 .  Do you propose doing something like this:

```
{key:value for key, value in snapshot_body.items() if value}
```

Also should i create a convenience function?

```
def _create_snapshot_body(comment=None, label=None, activity=None)
        snapshot_body = {"snapshotComment": comment,
                         "snapshotLabel": label,
                         "snapshotActivityId": activity}
        new_body = {key:value for key, value in snapshot_body.items() if value}
        return new_body)
```